eclipse-sirius/sirius-components
# Add Chart Widgets support in the Form description editor (WYSIWYG)

# Pull request template
Fix #1255 
## General purpose
What is the main goal of this pull request?

## Project management

## Architectural decision records (ADR)

## Dependencies

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- Hooks are properly typed:

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose

### Architecture

## Review

### How to test this PR?

- Create a new View model
- Create a new Form Description
- Create a new Form Description Editor on it
- Test that you can drag n drop PieChart and BarChart widgets from the left-hand side palette to the drop area.
- Check that you can change the widgets order.
- Check that you can delete them with the "delete" key.
